src/FastRouteRouter.php

```diff
@@ -547,18 +574,27 @@ private function cacheDispatchData(array $dispatchData)
         );
     }
 
+    /** @param FastRouteFoundResult $result */
     private function marshalMethodNotAllowedResult(array $result): RouteResult
     {
         $path           = $result[1];
-        $allowedMethods = array_reduce($this->routes, static function ($allowedMethods, $route) use ($path) {
-            if ($path !== $route->getPath()) {
-                return $allowedMethods;
-            }
+        $allowedMethods = array_reduce(
+            $this->routes,
+            /**
+             * @param list<string> $allowedMethods
```

Comment:
`list<non-empty-string>` surely?

Reply:
Yes, but that would require additional changes to `mezzio-router` - I feel like `list<string>` is sufficient considering the only use case is an exception message. To improve this type, I'll need to send another patch to `mezzio-router` and update and rebase here for negligible improvement.